### DISCUSSION TITLE
ESP32-2432S028R 2,8" -  Color Layout

### DISCUSSION THREAD

**Discussion by OctaMiner** [BitMaker-hub/NerdMiner_v2#273] [Q&A]
I have a quick, stupid question about the version for the 2.8" display and is it correct that the orange-blue display is displayed or is there something wrong with the display? Personally, I think the black layout is nicer.

But otherwise really good work, keep it up! :)

![NerdMiner V2 1 6 3 - 2 8 TFT](https://github.com/BitMaker-hub/NerdMiner_v2/assets/154564336/5c654fd4-06a2-43ed-921f-ae3e9fe42742)

**Comment by sany3001** [comment]
Just ordered same device, it is for very good price on Ali. I shall not forget to post here my own experience.

**Comment by BitMaker-hub** [comment]
Hi guys, did you flashed with last version? I think colors must be inverted.
I'll try and give back with a solution

**Reply by OctaMiner** [reply]
Yes, the software version is 1.6.3. But I have already tried the software on 1.6.2 and had the same colors on the display. Unfortunately, I'm not very familiar with the display and don't know how to calibrate it.

Of course, it could also be that it's the display itself, even though it's new.

**Reply by nitroxgas** [reply]
This seems to be a new version of that board.
I do not have one of these new boards to test.
I have compared mine with some pictures someone sent me, and they are different.

**Reply by OctaMiner** [reply]
But even if it were a new version of the board, wouldn't the software have to make it look like it does for all other users?

**Reply by nitroxgas** [reply]
Probably they have changed some display specs. It looks like the display isn't expecting the swap bytes that we use. But only with a board in hand to try out.

**Comment by sany3001** [comment]
I am asking if colors are such an issue?

**Reply by OctaMiner** [reply]
No, it's not a problem, it still works. I was just surprised that it was dark for everyone else and that it looked like that for me. Otherwise there could be an error and I would have to complain about it to the delivery company

**Comment by sany3001** [comment]
Understand, makes sense. I will share my experience as soon as I will get my piece.

That reminds me that I have to make donate to those great guys working on Nerdminer in the value of my last HW order.

**Comment by paelzer** [comment]
Hello

I found this workaround which helped me solving my color issues: https://github.com/Bodmer/TFT_eSPI/issues/2985

Two small changes I needed to do:

I added these lines below in the file **src/drivers/displays/esp23_2432s028r.cpp** into function **void esp32_2432S028R_Init(void)**.

```
tft.writecommand(ILI9341_GAMMASET); //Gamma curve selected
tft.writedata(2);
delay(120);
tft.writecommand(ILI9341_GAMMASET); //Gamma curve selected
tft.writedata(1);
```

I added these lines under **tft.setRotation(1);**

![image](https://github.com/BitMaker-hub/NerdMiner_v2/assets/16374477/3b142afb-4f78-4714-9ab5-2365f42a7984)

I also needed to add one line to the **platformio.ini** file:

`-DTFT_INVERSION_ON`

![image](https://github.com/BitMaker-hub/NerdMiner_v2/assets/16374477/4cf5f258-e579-42a2-b9de-0fba079e9c4d)


Before it looked ugly like this:
![image](https://github.com/BitMaker-hub/NerdMiner_v2/assets/16374477/02e9f2fd-982b-48f3-adfb-3d791ad7850a)

Now it's nice:
![image](https://github.com/BitMaker-hub/NerdMiner_v2/assets/16374477/6534d089-5656-4ec1-8fa8-7f73582d07f3)

I'm sure there is a better, or more professional way to fix this, but I don't have professional programming background. Maybe somebody more experienced here can have a look and improve this.

Hope this is helpful for somebody else as well.

Btw, this is my ESP32 2432S028 variant (with micro-USB and USB-C connector):
![image](https://github.com/BitMaker-hub/NerdMiner_v2/assets/16374477/36e8eacc-b5ee-4e07-8b7d-7e6fe008a741)


Boris

**Reply by ArminXXXXX** [reply]
where can I find the src folder? 2🤔

**Comment by sany3001** [comment]
I could not find this issue so I have created one new: https://github.com/BitMaker-hub/NerdMiner_v2/issues/285

I will close it now.

**Comment by paelzer** [comment]
Hi again

I just noticed, that I forgot to mention a second small change in the **platformio.ini** file in the above workaround. It's added now.

Boris

**Comment by BitMaker-hub** [comment]
Thanks guys, any who would like to make a PR for that reason.

**Comment by paelzer** [comment]
Hello

I have created a pull request regarding this topic. Please have a look :-)
https://github.com/BitMaker-hub/NerdMiner_v2/pull/295

Boris

**Comment by RainBitcoins** [comment]
I managed to change/add the code in visual basic but I'm still getting the old screen. This is my first time using VS, so do I just click on save? Then within the same folder I load those bins to my device or am I missing a step? Thx

**Reply by sany3001** [reply]
I have just pressed "Build" (tick) in left bottom corner and then "Upload" (arrow) next to that:

![image](https://github.com/BitMaker-hub/NerdMiner_v2/assets/31476550/c2e56405-3df0-43da-8a17-278e53270bb8)

From terminal output you will see if you have any problems or not. My problem was just proper serial port.

**Reply by RainBitcoins** [reply]
Ahh... Let me try, thank you.

**Comment by nitroxgas** [comment]
Hi everyone! 
I just received a board with the color inversion! 
I'll try to find a solution to keep only one code to the boards.  As soon as I find a way, I'll submit a PR.
The build directive TFT_INVERSION_ON works, but will force the deploy of two binary files. I'll try to find a solution to change the colors in run time.

**Reply by RainBitcoins** [reply]
Please... The other method with my lack of programming skill never worked for me.

**Reply by nitroxgas** [reply]
Done!
Pull Request to the DEV branch submitted!
https://github.com/BitMaker-hub/NerdMiner_v2/pull/397
Solution to the odd colors in new boards version (work on any variant), and some improvements. #397

**Reply by RainBitcoins** [reply]
Thank you but pardon my newbiness. Where do I get the updated bin? From the Github? Just download 1.6.3 again and it will have the new update with it?

**Reply by nitroxgas** [reply]
RainBitcoins, BitMaker will test and merge the code in the master branch and then release a binary in the web installer.
